### DISCUSSION TITLE
max_sessions_per_ip_per_user in nebula-metad.conf files

### DISCUSSION THREAD

**Discussion by porscheme** [vesoft-inc/nebula#5250] [Q&A]
@wey-gu 

Below is what the official dccs says about max_sessions_per_ip_per_use setting?
```
The maximum number of sessions that can be created with the same user and IP address.
```

Question: 

1. Is this limit for entire lifetime of user/ip combination? 
    For example, if a user try to make a new connection (over default 300 limit) in a weeks time, will it fail?

2. How can I make it unlimited?

Thanks for the help.
Porscheme

**Comment by wey-gu** [comment]
Sorry, maybe we could rephrase this(or you could actually help contribute that by clicking the ✏️ icon on top of the doc page).

It actually means the running(instead of created during a period of time) session per ip per user(combination, yes) is limited to `max_sessions_per_ip_per_use`, not for a lifetime, which by default is 300.

It's not recommended to be made unlimited as the session is persisted in metad(there is cost here), we may not be aware of any bad client/service creating too many sessions without hitting this limitation.

cc @abby-cyber 

You are always welcome @porscheme :)

**Reply by porscheme** [reply]
If I understand correctly, are you trying to say, this limit actually is...max open sessions per user/ip combination?

**Reply by wey-gu** [reply]
Yes, exactly, max open/active session per user/ip combination.

**Reply by porscheme** [reply]
Thanks for confirmation. I was hoping we should never hit max_sessions_per_ip_per_user limitation with the below settings in place.
session_idle_timeout_secs: 180
client_idle_timeout_secs: 3600
